python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -330,6 +330,78 @@ cdef class Column:
             c_result = make_column_from_scalar(dereference(slr.get()), size)
         return Column.from_libcudf(move(c_result))
 
+    @staticmethod
+    cdef Column _column_from_array_interface(
+        dict iface,
+        gpumemoryview data_view
+    ):
+        """
+        Construct a Column from array interface and associated device data.
+
+        Parameters
+        ----------
+        iface : dict
+            The parsed ``__array_interface__`` or ``__cuda_array_interface__`` dict.
```

Comment:
I don't think this dict is "parsed"? I think you're just passing it through.

I think I would prefer an interface where we accept the bits we need to interpret the data that have been provided in the `data_view` argument.

So something like:

```
@staticmethod
cdef Column from_gpumemoryview(gpumemoryview data, tuple shape, tuple strides, DataType dtype):
    if not is_c_contiguous(shape, strides, sizeof(dtype)):
        ....
    if len(shape) == 1:
        return Column(dtype, shape[0], data, None, 0, 0, [])
    elif len(shape) == 2:
       ...
    else:
        raise
```

WDYT?

Reply:
> I don't think this dict is "parsed"? I think you're just passing it through

Thanks this was left over when I thought I needed to split it out

> think I would prefer an interface where we accept the bits we need to interpret the data that have been provided in the data_view argument.

I like this